### PR TITLE
ensure newRow/newRecord is created according to table::getColNames()

### PR DESCRIPTION
**Pull request by yespire**
use case:
atlas->newRecord(mapper, data) // data may contain extra fields

### CODE REVIEW

**Comment by pmjones** [comment]
To make sure I understand, this just skips array elements that are not row columns -- yes?

**Comment by harikt** [comment]
yes.

**Comment by pmjones** [comment]
Good deal. Thanks!

File: src/Table/AbstractTable.php [hunk 1]
```diff
@@ -263,10 +263,16 @@ public function delete(RowInterface $row)
      * @return RowInterface
      *
      */
-    public function newRow(array $cols = [])
+    public function newRow(array $colsVals = [])
```
Comment (harikt) [inline src/Table/AbstractTable.php:266]: @pmjones I am @yespire have discussed something regarding the variable representation on Atlas. We have not opened an issue, just for not sure about it. For your consideration it is consistency of `$cols` and `$colVals` in some places. When it comes to `$cols` is it only column names , `$colVals` , column names with values. ( Associative array ) . If you have any inputs / think need to reopen an issue happy to discuss the same.

Thank you.

Comment (pmjones) [inline src/Table/AbstractTable.php:266]: I thought you might be @yespire :-)

The idea with $cols vs $colsVals is that $cols is for populating row data, and $colsVals is for queries. If you can think of a better convention, let me know.

Comment (harikt) [inline src/Table/AbstractTable.php:266]: > I thought you might be @yespire :-)

lol. no. I am not. I understand I wrote `I am` , what I was supposed to write was `I and` . Yes we also have not found a better naming / wording that is the reason we have been waiting to open an issue. Thank you for your consideration.

**Pull request merged**